### PR TITLE
Validate all file upload inputs

### LINKED ISSUE #1695: FileUpload validation only uses the status of the last question
https://github.com/seattle-uat/civiform/blob/main/universal-application-tool-0.0.1/app/assets/javascripts/validation.ts#L343

Multiple upload questions can be in a block but only one/first is validated.

### LINKED ISSUE #1696: Add focused testing for FileUpload Question
There is currently no focused test for the FileUpload question.

https://github.com/seattle-uat/civiform/blob/main/browser-test/src/address.test.ts

can be used as a template to test.

### PR DESCRIPTION
**Pull request by clouser2**
### Description
On-page file upload validation now validates ALL file upload questions in a block rather than assuming there is just one. Also fixes issue where Azure uploads threw an uncaught exception if a required file hadn't been uploaded (rather than showing a validation error to the user).

Note: Only one file upload is allowed per block by the admin controller. That said, there's not a strong need to bake that in at the validation level.

While here, I also:
  1) Renamed `validateFooQuestion` method names to `validateFooQuestions` to indicate they would validate all questions rather than just a single question.
  2) Added focused tests for file upload questions (which exposed a bug in the Azure file upload path, also fixed here).

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)
Fixes #1695; Fixes #1696

### CODE REVIEW

File: browser-test/src/file.test.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        await loginAsGuest(pageObject)
+        await selectApplicantLanguage(pageObject, 'English')
+
+        await applicantQuestions.applyProgram(programName)
+        await applicantQuestions.clickSkip()
+        await applicantQuestions.submitFromReviewPage(programName)
+      })
+    })
+  })
+
```
Comment (shanemc-goog) [inline browser-test/src/file.test.ts:132]: need a trailing newline it looks like?

**Review by shanemc-goog — CHANGES_REQUESTED** [review]
Looks good to me, any idea what's up with the test?

File: browser-test/src/file.test.ts (hunk 1 above)
Comment (clouser2) [reply browser-test/src/file.test.ts]: Fixed.

**Comment by clouser2** [comment]
> Looks good to me, any idea what's up with the test?

The failing test actually exposed a bug in the Azure file upload logic where `azure_upload.ts` was triggering prior to the code in `validation.ts`. This then led to an attempt to assume the file had always been uploaded, leading to an uncaught exception when that wasn't the case.

To fix this, I reworked how Azure uploads worked so that the logic in `validation.ts` is triggered first, which now correctly shows an error when attempting to click "Upload" without first having selected a file.

These changes are definitely worth taking another look. All tests are now passing and I've taken the pull request out of draft mode.

File: browser-test/src/file.test.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        [],
+        'file-upload-test-optional-q'
+      )
+      await adminPrograms.gotoAdminProgramsPage()
+      await adminPrograms.publishAllPrograms()
+
+      await logout(pageObject)
+    })
+
+    it('with missing file can be skipped', async () => {
```
Comment (goldblatt) [inline browser-test/src/file.test.ts:106]: can it though?

**Comment by shanemc-goog** [comment]
Thanks, I can look in the morning

**Comment by shanemc-goog** [comment]
How can I see the existing incorrect behavior that you're fixing in this PR?

**Review by shanemc-goog — APPROVED** [review]

File: browser-test/src/file.test.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  ApplicantQuestions,
+  loginAsAdmin,
+  loginAsGuest,
+  logout,
+  resetSession,
+  selectApplicantLanguage,
+  startSession,
+} from './support'
+
+describe('file upload applicant flow', () => {
```
Comment (shanemc-goog) [inline browser-test/src/file.test.ts:13]: I don't see a case for 2 upload questions.  Was that not actually doable after all?

The name question tests should have examples for you if that's helpful.

**Review by shanemc-goog — CHANGES_REQUESTED** [review]

File: browser-test/src/file.test.ts (hunk 3 above)
Comment (clouser2) [reply browser-test/src/file.test.ts:13]: The UI actually appears to disallow adding more than one file upload question to a single screen. I originally wrote a test which attempted this and ran into that failure.

Comment (shanemc-goog) [reply browser-test/src/file.test.ts:13]: Ah sorry, for some reason I thought Name was similar in only allowing one.

**Review by shanemc-goog — APPROVED** [review]

File: browser-test/src/file.test.ts (hunk 3 above)
Comment (clouser2) [reply browser-test/src/file.test.ts:13]: No worries at all. Resolving this.

**Pull request merged**